### DISCUSSION TITLE
Business rule that needs to run (only) once

### DISCUSSION THREAD

**Discussion by Chicagoan2016** [MarimerLLC/csla#3244] [Questions]
I have a business rule that needs to do some calculations. It also needs to increment a quantity inside 'Execute' method.
I want to make sure this business rule is run only once. A typical scenario is ,the main object is started by a user, she then forwards it to the manager and the manager could approve or send back to the user. I don't want the 'increment' portion to run every time the object is persisted.
Am I overthinking?
Kind regards.

**Comment by Chicagoan2016** [comment]
@rockfordlhotka Sir, when you get some down time.
Kind Regards

**Comment by rockfordlhotka** [comment]
On the surface this sounds like a workflow. Are you sure this is just one object? Or is it multiple objects for different stages of the workflow? Or maybe it is one object, but needs a property to indicate at which stage of the workflow it is current operating?

**Reply by Chicagoan2016** [reply]
@rockfordlhotka Thank you. 
I could add a property to this object to indicate the 'increment' is complete.

It got me thinking about other use case where we will use a business rule for tax calculations. Will the business rule for tax computation run every time we persist the object? 
Regards

**Reply by rockfordlhotka** [reply]
That's what I'd suggest - adding a property to indicate when the calculation is complete. It seems to me like this is a permanent status change for the type (and probably the underlying data).

Yes, by default the tax computation would run every time the object graph arrives on the logical server (so on every save). If the tax is already computed and you don't want to recompute, then I'd put a flag property in the object to indicate this state.
